src/verify-request/verify-request.ts

```diff
@@ -3,13 +3,21 @@ import compose from 'koa-compose';
 import {loginAgainIfDifferentShop} from './login-again-if-different-shop';
 import {verifyToken} from './verify-token';
 import {Options, Routes} from './types';
+import {DEFAULT_ACCESS_MODE} from '../auth';
 
 export default function verifyRequest(givenOptions: Options = {}) {
+  const {accessMode} = {
+    accessMode: DEFAULT_ACCESS_MODE,
+    ...givenOptions
+  };
```

Comment:
I've not seen this before - if `...givenOptions` contains `accessMode`, it overwrites the `accessMode` that's set earlier in the object?

Reply:
Yes, exactly, I believe it amounts to the same as
```ts
Object.assign({}, <defaults>, givenOptions);
```